### PR TITLE
mono: add support for *-musl

### PR DESCRIPTION
**Pull request by vjon**

### CODE REVIEW

**Comment by vjon** [comment]
This should also lay some groundwork for "armv7hf-musl" and "armv6hf-musl" support at some point in the future, although currently neither of those platforms is building properly.

**Comment by xtraeme** [comment]
Replace your conditional with:

```
case "$XBPS_TARGET_MACHINE" in
*-musl) ... ;;
esac
```

Don't forget to bump revision.

**Comment by vjon** [comment]
Done and done!

File: srcpkgs/mono/template [hunk 1]
```diff
@@ -16,6 +16,15 @@ license="GPL-2, LGPL-2, MPL, MIT"
 distfiles="http://download.mono-project.com/sources/mono/$pkgname-$version.tar.bz2"
 checksum=eaf5bd9d19818cb89483b3c9cae2ee3569643fd621560da036f6a49f6b3e3a6f
 
+_musl_args=" --disable-boehm --without-sigaltstack"
+case "$XBPS_TARGET_MACHINE" in
+	*-musl) configure_args+=$_musl_args ;;
```
Comment (xtraeme) [inline srcpkgs/mono/template]: Please get rid of this local var, simply use `configure_args+=" ... "`.

**Comment by pullmoll** [comment]
This patch fixes cross compiling for armv[67]hf-musl: http://sprunge.us/AFgK
Note that the build lists a number of shared libraries missing from common/shlibs:
```
=> WARNING: mono-4.0.1.44_2: libmono-profiler-cov.so.0 not found in common/shlibs!
=> WARNING: mono-4.0.1.44_2: libmono-profiler-aot.so.0 not found in common/shlibs!
=> WARNING: mono-4.0.1.44_2: libmono-profiler-iomap.so.0 not found in common/shlibs!
=> WARNING: mono-4.0.1.44_2: libmono-profiler-log.so.0 not found in common/shlibs!
=> WARNING: mono-4.0.1.44_2: libikvm-native.so not found in common/shlibs!
=> WARNING: mono-4.0.1.44_2: libMonoPosixHelper.so not found in common/shlibs!
=> WARNING: mono-4.0.1.44_2: libMonoSupportW.so not found in common/shlibs!
```

**Comment by vjon** [comment]
Added the patch, thanks @pullmoll for that! Also updated shared libraries in the map file.

**Pull request merged**